tests/backend/test_bosonic_backend.py

```diff
@@ -0,0 +1,566 @@
+        assert state.covs().shape == (num_weights_high, 2, 2)
+
+        if alpha != 0:
+            assert num_weights_low < num_weights_high
+        else:
+            assert num_weights_low == num_weights_high
+
+    @pytest.mark.parametrize("alpha", ALPHA_VALS)
+    @pytest.mark.parametrize("phi", PHI_VALS)
+    def test_cat_state_wigners(self, alpha, phi):
+        r"""Checks that the real and complex cat state representations
+        have the same Wigner functions as the cat state from the Fock
+        backend."""
+        x = np.linspace(-2 * alpha, 2 * alpha, 100)
+        p = np.linspace(-2 * alpha, 2 * alpha, 100)
+
+        prog_complex_cat = sf.Program(1)
+        with prog_complex_cat.context as qc:
+            sf.ops.Catstate(alpha, phi) | qc[0]
+
+        prog_real_cat = sf.Program(1)
+        with prog_real_cat.context as qr:
+            sf.ops.Catstate(alpha, phi, desc="real", D=10) | qr[0]
+
+        backend_complex = bosonic.BosonicBackend()
+        backend_complex.run_prog(prog_complex_cat)
+        wigner_complex = backend_complex.state().wigner(0, x, p)
+
+        backend_real = bosonic.BosonicBackend()
+        backend_real.run_prog(prog_real_cat)
+        wigner_real = backend_real.state().wigner(0, x, p)
+
+        prog_cat_fock = sf.Program(1)
+        with prog_cat_fock.context as qf:
+            if alpha != 0:
+                sf.ops.Catstate(alpha, phi) | qf[0]
+            else:
+                sf.ops.Vacuum() | qf[0]
+        eng = sf.Engine("fock", backend_options={"cutoff_dim": 20})
+        results = eng.run(prog_cat_fock)
+        wigner_fock = results.state.wigner(0, x, p)
+
+        assert np.allclose(wigner_complex, wigner_real, rtol=1e-3, atol=1e-6)
+        assert np.allclose(wigner_complex, wigner_fock, rtol=1e-3, atol=1e-6)
+        assert np.allclose(wigner_fock, wigner_real, rtol=1e-3, atol=1e-6)
+
+    @pytest.mark.parametrize("alpha", ALPHA_VALS)
+    def test_cat_state_parity(self, alpha):
+        r"""Checks that the real and complex cat state representations
+        yield the correct parity."""
+        # for phi = 0, should yield parity of 1
+        prog_complex_cat = sf.Program(1)
+        with prog_complex_cat.context as qc:
+            sf.ops.Catstate(alpha) | qc[0]
+
+        prog_real_cat = sf.Program(1)
+        with prog_real_cat.context as qr:
+            sf.ops.Catstate(alpha, desc="real") | qr[0]
+
+        backend_complex = bosonic.BosonicBackend()
+        backend_complex.run_prog(prog_complex_cat)
+        state_complex = backend_complex.state()
+        parity_complex = state_complex.parity_expectation([0])
+
+        backend_real = bosonic.BosonicBackend()
+        backend_real.run_prog(prog_real_cat)
+        state_real = backend_real.state()
+        parity_real = state_real.parity_expectation([0])
+
+        assert np.allclose(parity_complex, 1)
+        assert np.allclose(parity_real, 1)
+
+        # for phi = 1, should yield parity of -1 unless alpha == 0
```

Comment:
Seems to be a separate test case, could separate. Also taking `0` out of `ALPHA_VALS` for this test will help eliminate the `if alpha != 0:` condition.

Reply:
I would like to test the case where alpha = 0, though.

Reply:
It also future-proofs against someone adding 0 to the list of alphas to test.